File: hello-world/hello.lisp.lua
#!/usr/bin/env lua
;--[[ Code visible to Common Lisp

(defun println (x) (princ x) (princ #\Newline) nil)
(println "Hello from Common Lisp!")

#|| Code visible to Lua ]]

function void(...) end
function println(...) print(...) return void end
println("Hello from Lua!")

-- Code visible to Common Lisp and Lua ||#

(println "Hello from Common Lisp and Lua!")

File: hello-world/hello_lisp.rb
#!/usr/bin/env ruby
#| code visible to Ruby

puts "Hello from Ruby!"

=begin code visible to Common Lisp |#

(defun puts (x) (princ x) (princ #\Newline))
(puts "Hello from Common Lisp!")

#|
=end # code visible to Common Lisp and Ruby |#

(puts "Hello from Common Lisp and Ruby!")
